gateway/src/apicast/policy/metrics/metrics.lua

```diff
@@ -0,0 +1,113 @@
+local _M = require('apicast.policy').new('Metrics')
```

Comment:
We might want to call this Nginx Metrics ? Probably need some input from @3scale/product.

Several policies will be exposing metrics and this one is really focused on internal nginx stuff like shared dictionaries, connections etc.